main/app.c

```diff
@@ -2180,51 +2296,243 @@ int ast_app_group_split_group(const char *data, char *group, int group_max, char
 	return res;
 }
 
-int ast_app_group_set_channel(struct ast_channel *chan, const char *data)
+/* NOTE: Ideally we would also be able to use direct pointers to the group meta, so we don't have to search */
+int ast_app_group_remove_meta_ifneeded(const char *group, const char *category)
 {
-	int res = 0;
-	char group[80] = "", category[80] = "";
-	struct ast_group_info *gi = NULL;
-	size_t len = 0;
+	struct ast_group_meta *gmi = NULL; /*!< Group metadatas	  */
 
-	if (ast_app_group_split_group(data, group, sizeof(group), category, sizeof(category))) {
-		return -1;
+	struct varshead *headp;
+	struct ast_var_t *vardata;
+
+	int destroy = 0;
```

Comment:
Technically, this variable is not needed, gmi being non-NULL after the loop provides the same guarantee

Reply:
Fixed